mlrun/api/schemas/model_endpoints.py

```diff
@@ -28,36 +31,74 @@ class ModelMonitoringStoreKinds:
 
 
 class ModelEndpointMetadata(BaseModel):
-    project: Optional[str]
-    labels: Optional[dict]
-    uid: Optional[str]
+    project: Optional[str] = ""
+    labels: Optional[dict] = {}
+    uid: Optional[str] = ""
 
     class Config:
         extra = Extra.allow
 
+    @classmethod
+    def from_dict(cls, endpoint_dict: dict, json_parse_values: typing.List = None):
+        """Create a ModelEndpointMetadata object from an endpoint dictionary
+
+        :param endpoint_dict:     Model endpoint dictionary.
+        :param json_parse_values: List of dictionary keys with a JSON string value that will be parsed into a
+                                  dictionary using json.loads().
+        """
+        new_object = cls()
+        if json_parse_values is None:
+            json_parse_values = [model_monitoring_constants.EventFieldType.LABELS]
+        # Replace the key name endpoint_id with uid as required by the object schema
+        if model_monitoring_constants.EventFieldType.ENDPOINT_ID in endpoint_dict:
+            endpoint_dict[
+                model_monitoring_constants.EventFieldType.UID
+            ] = endpoint_dict.pop(model_monitoring_constants.EventFieldType.ENDPOINT_ID)
+        return _mapping_attributes(
+            base_model=new_object,
+            flattened_dictionary=endpoint_dict,
+            json_parse_values=json_parse_values,
+        )
+
 
 class ModelMonitoringMode(str, enum.Enum):
```

Comment:
If you for some reason wish to inherit both `Enum` and `str` you have `enum.StrEnum` but I don't see a reason why?
```suggestion
class ModelMonitoringMode(enum.Enum):
```

Reply:
If `ModelMonitoringMode` doesn't inherit the `str` then you'll get `ModelMonitoringMode is not JSON serializable` when process the `ModelEndpointSpec.monitoring_mode` value.
Therefore we have to keep the `str` as well.